sessionctx/variable/session.go

```diff
@@ -722,6 +754,38 @@ func (s *SessionVars) IsAutocommit() bool {
 	return s.GetStatusFlag(mysql.ServerStatusAutocommit)
 }
 
+// IsReadConsistencyTxn if true it means the transaction is an read consistency (read committed) transaction.
+func (s *SessionVars) IsReadConsistencyTxn() bool {
+	if s.TxnCtx.Isolation != "" {
+		return s.TxnCtx.Isolation == ast.ReadCommitted
+	}
+	if s.txnIsolationLevelOneShot.state == oneShotUse {
+		s.TxnCtx.Isolation = s.txnIsolationLevelOneShot.value
```

Comment:
When will this value be reset?

Reply:
When the session starts a new transaction, it will create a new TxnCtx. It works like StmtCtx for each statement.